crates/wysiwyg/src/dom/nodes/container_node.rs

```diff
@@ -366,7 +366,18 @@ where
         let Some(first_child) = self.children.get_mut(0) else {
             return false;
         };
-        first_child.add_leading_zwsp()
+        match first_child {
+            DomNode::Container(c) => c.add_leading_zwsp(),
+            DomNode::Zwsp(_) => false,
+            _ => {
+                if self.handle().is_set() {
+                    self.insert_child(0, DomNode::new_zwsp());
+                } else {
+                    self.children.insert(0, DomNode::new_zwsp());
+                }
```

Comment:
Wrote this code, which is not really required on current context because we are on a detached state from Dom. So we are fine with unset handles. 
I wonder though if what we want as a solution is this kind of thing or rather setting up a fake handle when transforming items in detached state.